src/include/mpir_comm.h

```diff
@@ -258,9 +259,26 @@ int MPIR_Comm_split_filesystem(MPI_Comm comm, int key, const char *dirname, MPI_
 #define MPIR_Comm_rank(comm_ptr) ((comm_ptr)->rank)
 #define MPIR_Comm_size(comm_ptr) ((comm_ptr)->local_size)
 
-/* Communicator info hint functions */
-typedef int (*MPIR_Comm_hint_fn_t) (MPIR_Comm *, MPIR_Info *, void *);
-int MPIR_Comm_register_hint(const char *hint_key, MPIR_Comm_hint_fn_t fn, void *state);
+/* Communicator info hint */
+#define MPIR_COMM_HINT_TYPE_BOOL 0
+#define MPIR_COMM_HINT_TYPE_INT  1
+
+#define MPIR_COMM_HINT_ATTR_LOCAL 0x1
```

Comment:
Can you add a code comment to say why we need this?

Reply:
To mark whether the hint need be the same across communicator. 

Yes, I will add a code comment.